src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,52 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertStringsToTransactionsList(List<String> lines) {
+        if (lines == null) {
+            throw new NullPointerException("Input parameter cannot be null");
```

When `lines` is null you throw `new NullPointerException(...)`. The checklist requires validating null parameters and handling invalid inputs, and also states: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and include the cause [CHECKLIST ITEM #8]." Please replace this with a RuntimeException (or IllegalArgumentException wrapped in a RuntimeException) with an informative message indicating the parameter and context, e.g., `throw new RuntimeException("Input lines list cannot be null")`.